setzer/dialogs/preferences/pages/page_build_system.py

```diff
@@ -98,7 +98,18 @@ def setup_latex_interpreters(self):
                 self.view.option_latex_interpreter[interpreter].set_active(interpreter == self.settings.get_value('preferences', 'latex_interpreter'))
             for interpreter in self.latex_interpreters:
                 self.view.option_latex_interpreter[interpreter].connect('toggled', self.preferences.on_interpreter_changed, 'latex_interpreter', interpreter)
+            
+            self.view.option_latex_interpreter['tectonic'].connect('toggled', self.on_use_tectonic_toggled)
+            self.view.latexmk_enable_revealer.set_reveal_child(not self.view.option_latex_interpreter['tectonic'].get_active())
+            self.view.shell_escape_revealer.set_reveal_child(not self.view.option_latex_interpreter['tectonic'].get_active())
 
+    def on_use_tectonic_toggled(self, button):
+        self.view.latexmk_enable_revealer.set_reveal_child(not button.get_active())
+        self.settings.set_value('preferences', 'use_latexmk', not button.get_active())
```

Comment:
i don't think this should be in there. switching back to another build system, this preference might be helpful and should not be overwritten.

Reply:
do you mean hiding the latexmk option, or changing the setting? or both?